Assets/Scripts/Boat/BoatCapacity.cs

```diff
@@ -34,19 +87,24 @@ void Start()
     /// <param name="amount">By how many units should the maximum capacity be increased.</param>
     public void IncreaseCapacity(int amount = 1)
     {
-        currentCapacity += amount;
+        CurrentCapacity += amount;
+    }
+
+    public void IncreaseCapacity()
+    {
+        IncreaseCapacity(capacityRestoredOnSuccessfulFerry);
     }
 
     private int DecreaseCapacity(int amount = 1)
     {
-        currentCapacity = currentCapacity - amount;
-        if (currentCapacity == 0)
+        CurrentCapacity = CurrentCapacity - amount;
+        if (CurrentCapacity == 0)
         {
             OnBoatDestroyed?.Invoke();
         }
-        else if (currentCapacity < 0)
+        else if (CurrentCapacity < 0)
```

Comment:
I'm a bit confused by this logic. If we have one capacity and amount is set to 2, then we don't die, but if it's set to 1 we do?